src/caffe/layers/image_data_layer.cpp

```diff
@@ -46,6 +46,9 @@ void ImageDataLayer<Dtype>::DataLayerSetUp(const vector<Blob<Dtype>*>& bottom,
     lines_.push_back(std::make_pair(line.substr(0, pos), label));
   }
 
+  // Check that image list is not empty.
+  CHECK_GT(lines_.size(), 0);
+
```

Comment:
It's better practice, more idiomatic.
It doesn't matter for a vector, but for a list it does, in C++98, the complexity `list.size()` is not guaranteed to be constant.